### PR TITLE
imx6qdl-pico: Add QCA9377 Wifi support

### PR DESCRIPTION
**Pull request by fabioestevam**
Add QCA9377 Wifi support.

Tested on a imx6dl-pico-pi board by doing:

ifconfig wlan0 up
iw dev wlan0 connect ACCESSPOINTNAME
wpa_passphrase ACCESSPOINTNAME > /etc/wpa.conf
(enter the wifi password and press enter)
wpa_supplicant -Dwext -iwlan0 -c /etc/wpa.conf &
udhcpc -i wlan0
ping google.com

Signed-off-by: Fabio Estevam <festevam@gmail.com>

### CODE REVIEW

File: conf/machine/imx6qdl-pico.conf [hunk 1]
```diff
@@ -44,8 +45,13 @@ MACHINE_ESSENTIAL_EXTRA_RDEPENDS += " \
 
 MACHINE_EXTRA_RRECOMMENDS += " \
     bcm4339-nvram-config \
+    wpa-supplicant \
```
Comment (otavio) [inline conf/machine/imx6qdl-pico.conf:48]: This doesn't seem right as it is not required for the machine to work. Also, a device can connect to an open wifi without it. This should be added to the user image.

**Review by otavio — CHANGES_REQUESTED** [review]

**Comment by fabioestevam** [comment]
Hi Otavio,

On Sun, Jun 6, 2021 at 7:09 PM Otavio Salvador ***@***.***> wrote:

> In conf/machine/imx6qdl-pico.conf:
>
> > @@ -44,8 +45,13 @@ MACHINE_ESSENTIAL_EXTRA_RDEPENDS += " \
>
>  MACHINE_EXTRA_RRECOMMENDS += " \
>      bcm4339-nvram-config \
> +    wpa-supplicant \
>
> This doesn't seem right as it is not required for the machine to work. Also, a device can connect to an open wifi without it. This should be added to the user image.

I followed the example at conf/machine/imx7d-remarkable2.conf where
wpa_supplicat is also passed.

I thought it would be nice to get wpa_supplicant added by default so I
can easily connect to a protected network.

 When you say "This should be added to the user image", what does that
mean exactly?

Thanks

**Comment by otavio** [comment]
We then need to remove it from the `imx7d-remarkable2` machine.

The idea of machine-specific packages on the machine definition file is to enable essential machine features, and the WPA is not essential for WiFi, or the machine, to work. We can use conman, NetworkManager, or others to connect to WiFi.

That kind of packages ought to be added inside the product-specific image and not in the BSP. Imagine if a product does not require WiFi, so the `wpa-supplicant` would bloat the image for no good reason.

**Comment by otavio** [comment]
Fixed in #144

**Pull request closed without merging**